client/components/Menubar/contexts.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { createContext, RefObject } from 'react';
+
+export const ParentMenuContext = createContext<string>('none');
+
+export const MenuOpenContext = createContext<string>('none');
+
+interface MenubarContextType {
+  createMenuHandlers: (
+    id: string
+  ) => {
+    onMouseOver: (e: React.MouseEvent) => void;
```

Comment:
I'm not strongly opinionated about this so feel free to just acknowledge and ignore, but I noticed that the logic changes a bit in this file where the factory methods (e.g. createMenuHandlers) previously would return an empty dict {} but now return dicts where the value is a function that doesn't do anything. I think if there's logic further down the line that checks somehting like menuHandlers.onMouseOver, the return would be different. in the past where it would be `undefined` now it would return `() => {}`. if you added the null functions to avoid type errors, we could just make the handlers optional? just wanted to flag it in case

Reply:
LGTM!! ty!